COPYING

```diff
@@ -1,4 +1,5 @@
 Copyright (c) 2006-2020, Salvatore Sanfilippo
+Copyright (c) 2020-2024, placeholderkv
```

Comment:
We had a discussion, https://github.com/redis/redis/issues/9236, in the past. I would ask you wait until  we finalize the name to update it here though.

Reply:
My vote on OpenKV

Reply:
To clarify my comment: copyright is an important legal concept and is not something obtained by simply forking a git repo - first, some copyright-able original work must be created.  No doubt this will happen in due course though, and at that time you'll need to make some decisions.  Also, copyright cannot be backdated arbitrarily as in this PR - it related specifically to the time the original work was created.

Reply:
```suggestion
Copyright (c) 2024, placeholderkv contributors
```
I suppose would be the more correct approach then? We talked about this before forking as well, but nobody cared too much about the licenses at the time. We do have some changes here already, it's not like nothing has happened.

Reply:
Yes, I've seen that done in some projects I've work on.  It could be seen as a watering-down of the copyright though, especially if the "placeholderkv contributors" group is not clearly defined or changes over time.

| We do have some changes here already, it's not like nothing has happened.

Not every change is copyright-able - it needs to have some aspect of originality to it.  There are also lots of rules-of-thumb I've seen over the years (some from lawyers, some from developers paraphrasing, or looking for rules to follow that don't exist, so YMMV) - e.g. "small changes", "less than 20 lines", "less than 100 lines", ... but to the best of my knowledge there is no such rule, it is the originality aspect that is key when it comes to copyright (AIUI, IANAL).

BTW, one widely established approach in terms of tracking attribution is that of the Linux kernel and git projects, where Signed-off-by is used along with a file in the repo explaining what that means to new contributors, e.g.

https://git.kernel.org/pub/scm/git/git.git/tree/Documentation/SubmittingPatches?id=HEAD 

See the section from 'Developer's Certificate of Origin' onward.  Definitely worth thinking about following that strategy here.

Finally, in your redis issue you linked you mentioned maybe not explicitly having the copyright/SPDX notice in each file - the advice I've been given from lawyers is to always have those two headers (both copyright and license) at the start of every source file.

Reply:
> Yes, I've seen that done in some projects I've work on. It could be seen as a watering-down of the copyright though, especially if the "placeholderkv contributors" group is not clearly defined or changes over time.

The guidance from other folks was not that it was a group, just that it's everyone that has contributed. 

> See the section from 'Developer's Certificate of Origin' onward. Definitely worth thinking about following that strategy here.

This is on our roadmap, I was looking at it tonight and will probably enable it tomorrow. 

> Finally, in your redis issue you linked you mentioned maybe not explicitly having the copyright/SPDX notice in each file - the advice I've been given from lawyers is to always have those two headers (both copyright and license) at the start of every source file.

I believe my point was that I was advised it was okay to skip touching existing files if they already had the full copyright, if you have a different position, I don't see any reason not to update them now (as the license files are annoyingly verbose). 

> Not every change is copyright-able - it needs to have some aspect of originality to it. There are also lots of rules-of-thumb I've seen over the years (some from lawyers, some from developers paraphrasing, or looking for rules to follow that don't exist, so YMMV) - e.g. "small changes", "less than 20 lines", "less than 100 lines", ... but to the best of my knowledge there is no such rule, it is the originality aspect that is key when it comes to copyright (AIUI, IANAL).

Point taken. I don't think there is anything wrong with adding it though, it just wouldn't hold up.

Reply:
"wouldn't hold up" -> "may not hold up" in legalese.  Never commit yourself either way - it's the first rule of lawyer fight club.  :)

Reply:
yeah, I like the Developer's Certificate of Origin I think a few other projects in the past I contributed to I had to add the `Signed-off-by: ` (and now I know where they got that from.)

maybe we should have a issue going over how to handle adding a Developer's Certificate of Origin or is there one already?

Reply:
>"wouldn't hold up" -> "may not hold up" in legalese. Never commit yourself either way - it's the first rule of lawyer fight club. :)

I'm so glad we have people who know this better than I do here :)

I've also added the DCO check, and playing around to see how it behaves.